src/sentry/rules/processing/delayed_processing.py

```diff
@@ -320,3 +319,17 @@ def apply_delayed(project_id: int) -> None:
                 rule, groupevent, notification_uuid, rule_fire_history
             ).values():
                 safe_execute(callback, groupevent, futures, _with_transaction=False)
+
+    # Step 8: Clean up Redis buffer data
+    for rule, groups in rules_to_groups.items():
+        for group in groups:
+            buffer.delete_hash(
+                model=Project,
+                filters={"project_id": project_id},
+                field=f"{rule}:{group}",
+            )
```

Comment:
As a potential follow up, we might want to implement a batch method for this that builds up a pipeline and sends these in one request.

Reply:
Hmm maybe I can just modify `delete_hash` to be like `add_hash_to_delete_pipeline` (idk maybe keep the name the same) but basically just not do `pipe.execute()` and have another function that only executes?

Reply:
Doesn't seem like that much work, I can just add that change into this PR

Reply:
Hmm, I don't know if we want to store the `pipeline` locally in the class, there might be concurrency problems there if we have multiple threads.

It might be easier to just have `buffer.delete_hash_bulk(model=Project, filters={"project_id": project_id}, fields=[<list of all rule and group strings>]`